dev-packages/application-manager/src/application-package-manager.ts

```diff
@@ -78,7 +78,13 @@ export class ApplicationPackageManager {
     }
 
     startElectron(args: string[]): cp.ChildProcess {
-        const { mainArgs, options } = this.adjustArgs([this.pck.frontend('electron-main.js'), ...args]);
+        const { mainArgs, options } = this.adjustArgs([
+            // If possible, pass the project root directory to electron rather than the script file so that Electron
+            // can determine the app name. This requires that the package.json has a main field.
+            this.pck.pck.main ? this.pck.projectPath : this.pck.frontend('electron-main.js'),
```

Comment:
We could print out a warning too with code which should be added?

Reply:
I've added a warning message now.